RickAndMorty-iOS/Coordinators/MainCoordinator.swift

```diff
@@ -0,0 +1,152 @@
+        demoViewController.coordinator = self
+        demoNavController.navigationBar.backgroundColor = .systemBackground
+        demoNavController.pushViewController(demoViewController, animated: false)
+    }
+    // MARK: -
+
+    init(window: UIWindow) {
+        self.window = window
+    }
+
+    func start() {
+        // Configure tab bar.
+        tabBarController.tabBar.backgroundColor = .systemBackground
+        window.backgroundColor = .systemRed
+
+        let navBarAppearance = customNavBarAppearance()
+
+        // Add navigation controllers to the tab bar and set the title and icon for each tab.
+        tabBarController.addChild(characterNavController)
+        characterNavController.tabBarItem.image = UIImage(systemName: "person.text.rectangle")
+        characterNavController.tabBarItem.title = "Characters"
+        characterNavController.navigationBar.standardAppearance = navBarAppearance
+
+        tabBarController.addChild(locationNavController)
+        locationNavController.tabBarItem.image = UIImage(systemName: "map")
+        locationNavController.tabBarItem.title = "Locations"
+        locationNavController.navigationBar.standardAppearance = navBarAppearance
+
+        tabBarController.addChild(episodeNavController)
+        episodeNavController.tabBarItem.image = UIImage(systemName: "tv")
+        episodeNavController.tabBarItem.title = "Episodes"
+        episodeNavController.navigationBar.standardAppearance = navBarAppearance
+
+        tabBarController.addChild(searchNavController)
+        searchNavController.tabBarItem.image = UIImage(systemName: "magnifyingglass")
+        searchNavController.tabBarItem.title = "Search"
+        searchNavController.navigationBar.standardAppearance = navBarAppearance
+
+        // Add CharactersViewController to the character navigation controller.
+        let charactersViewController = CharactersViewController()
+        charactersViewController.coordinator = self
+        characterNavController.pushViewController(charactersViewController, animated: false)
+
+        // Add LocationsViewController to the location navigation controller.
+        let locationsViewController = LocationsViewController()
+        locationsViewController.coordinator = self
+        locationNavController.pushViewController(locationsViewController, animated: false)
+
+        // Add EpisodesViewController to the episode navigation controller.
+        let episodesViewController = EpisodesViewController()
+        episodesViewController.coordinator = self
+        episodeNavController.pushViewController(episodesViewController, animated: false)
+
+        // Add SearchViewController to the search navigation controller.
+        let searchViewController = SearchViewController()
+        searchViewController.coordinator = self
+        searchNavController.pushViewController(searchViewController, animated: false)
+
+        // MARK: - Setup demo tab for experiment, remove this from production.
```

Comment:
Release code should not have commented code or dead code. Although it is ok to have them while in dev